testflinger_agent/__init__.py

```diff
@@ -26,9 +31,127 @@
 logger = logging.getLogger(__name__)
 
 
+class LoopTimer(Timer):
+    """Loop thread timer (interval)"""
+
+    def run(self):
+        while not self.finished.wait(self.interval):
+            self.function(*self.args, **self.kwargs)
+
+
+class ReqLogger(Thread):
```

Comment:
hmm, I need to take a little more time to understand this threaded log handler. I was thinking that you originally intended to break that out into a separate PR?

So I haven't had a chance to try it yet, but my initial thoughts are that this is a little complicated, and I'm wondering if we could simplify it. To me, it seems like the log handler here is mostly like a buffer - we just want to buffer the output rather than sending it right away. The time component of it only needs to be triggered when something enters the buffer for the first time. That piece may need to be a thread, but all it needs to do is flush the buffer once it times out. So I'm wondering whether we need to create the handler inside a thread that persists for the duration of the agent, or whether we could simply create a thread whenever needed to handle the buffer timeout? Just a thought.